src/views/clusteroverview/overview/components/status-card/utils/hooks/useAlerts.ts

```diff
@@ -0,0 +1,159 @@
+      ? getRangeVectorSearchParams(endTime, samples, timespan)
+      : new URLSearchParams();
+  Object.entries(params).forEach(
+    ([key, value]) => value && searchParams.append(key, value.toString()),
+  );
+  return searchParams;
+};
+
+export const getPrometheusURL = (
+  props: PrometheusURLProps,
+  basePath: string = props.namespace ? PROMETHEUS_TENANCY_BASE_PATH : PROMETHEUS_BASE_PATH,
+): string => {
+  if (props.endpoint !== PrometheusEndpoint.RULES && !props.query) {
+    return '';
+  }
+  const params = getSearchParams(props);
+  return `${basePath}/${props.endpoint}?${params.toString()}`;
+};
+
+type PrometheusURLProps = {
+  endpoint: PrometheusEndpoint;
+  endTime?: number;
+  namespace?: string;
+  query?: string;
+  samples?: number;
+  timeout?: string;
+  timespan?: number;
+};
+
+type Group = {
+  rules: PrometheusRule[];
+  file: string;
+  name: string;
+};
+
+export type PrometheusRulesResponse = {
+  data: {
+    groups: Group[];
+  };
+  status: string;
+};
+
+const getAlertsAndRules = (
+  data: PrometheusRulesResponse['data'],
+): { alerts: Alert[]; rules: Rule[] } => {
+  // Flatten the rules data to make it easier to work with, discard non-alerting rules since those
+  // are the only ones we will be using and add a unique ID to each rule.
+  const groups = get(data, 'groups') as PrometheusRulesResponse['data']['groups'];
+  const rules = flatMap(groups, (g: Group) => {
+    const addID = (rule: PrometheusRule): Rule => {
+      const key = [
+        g.file,
+        g.name,
+        rule.name,
+        rule.duration,
+        rule.query,
+        ...Object.entries(rule.labels).map((k, v) => `${k}=${v}`),
+      ].join(',');
+      return { ...rule, id: `${key}-monitoring-salt` };
+    };
+
+    const alertingFilter = (r: PrometheusRule) => r.type === 'alerting';
+    return g.rules.filter(alertingFilter).map(addID);
+  });
+
+  // Add `rule` object to each alert
+  const alerts = flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
+
+  return { alerts, rules };
+};
+
+const useAlerts = () => {
+  const [alertsLoading, setAlertsLoading] = React.useState<boolean>(false);
+  const [alertsLoaded, setAlertsLoaded] = React.useState<boolean>(false);
+  const [alertsError, setAlertsError] = React.useState<Error>(null);
+  const [alertsData, setAlertsData] = React.useState([]);
+
+  React.useEffect(() => {
+    const { prometheusBaseURL } = (window as any).SERVER_FLAGS;
+
+    const pollers = {};
+    const pollerTimeouts: { [key: string]: NodeJS.Timeout } = {};
+
+    if (prometheusBaseURL) {
+      const alertsKey = 'alerts';
+      // const rulesKey = 'rules';
+      setAlertsLoading(true);
+
+      const url = getPrometheusURL({ endpoint: PrometheusEndpoint.RULES });
+
+      const poller = (): void => {
+        consoleFetchJSON(url)
+          .then(({ data }) => {
+            const { alerts } = getAlertsAndRules(data);
+            setAlertsLoaded(!!alerts);
+            setAlertsData(alerts);
```

Comment:
@pcbailey I can't find where you filter only alerts relevant to virtualization, what am I missing ?

Reply:
You're not missing anything. =) They're not filtered yet because I don't know how to filter them. It's an issue that fell through the cracks in the static plugin version. I need to talk to Shirly when she gets back to find out the best way to filter them.